### DISCUSSION TITLE
Issue with GPS location

### DISCUSSION THREAD

**Discussion by bluecasita** [immich-app/immich#8868] [Q&A]
### The bug

I'm having a weird issue with several photos and their GPS info...

I live in Houston, TX.  I have many photos taken here.  However, in many cases, while the GPS pin on the map shows exactly the right place, the location is listed incorrectly.  I'm seeing a location of Hillshire Village instead of Houston, and I am 100% sure, Houston is the correct location.  Hillshire Village is 10 miles away.

Is there a way to fix this for all my photos.  As I said, GPS coordinates are correct but the location lookup is wrong.

Thanks in advance for any help!

### The OS that Immich Server is running on

Docker

### Version of Immich Server

v1.101.0

### Version of Immich Mobile App

N/A

### Platform with the issue

- [ ] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
N/A
```

### Your .env content

```Shell
N/A
```

### Reproduction steps

```bash
1.N/A
2.
3.
...
```

**Comment by bo0tzz** [comment]
This is probably working as expected. The geocoding is pretty rudimentary, and just based on what the closest point in the database is rather than on precise administrative boundaries or such.

**Reply by bluecasita** [reply]
I figured as much. Is there any way to manually update that database without changing the GPS coordinates?

I also noticed that there is a setting for the accuracy of the GPS. I remember seeing an option to enter a 1, 2 or 3. I wonder if changing that might help. I have it set to the most accurate right now.

**Reply by bo0tzz** [reply]
> Is there any way to manually update that database without changing the GPS coordinates?

Not at the moment, no

> there is a setting for the accuracy of the GPS.

Where are you seeing this? We used to have a setting for this quite a while ago, but it got removed.

**Reply by bluecasita** [reply]
To be honest I used this guide to install Immich.  
https://mariushosting.com/how-to-install-immich-on-your-synology-nas/ and it provided the attached stack.env file.  Maybe it was outdated...
[stack.env.txt](https://github.com/immich-app/immich/files/15018579/stack.env.txt)

If you could let me know what is wrong, I will fix it.  Thanks so much for your help...

**Reply by bluecasita** [reply]
I was able to update the environment variables just to use the ones in the documentation...

![image](https://github.com/immich-app/immich/assets/154034577/520c5ff8-384b-4e42-aed9-74d757012c07)

Still didn't fix my issue though :) It's so odd how the geocoding thinks that a location right in the middle of Houston is actually a minute village 10 miles away...

**Comment by ronnicek** [comment]
Hi, I had same issue, and I found out that city didn't had population on geonames.org from where Immich is buidling the database. I see that city is there after another export (https://download.geonames.org/export/dump/), but when I tried to redeploy Immich, it was not there, so I assume it's downloaded on container build? But I do not see anything in the code where this is happening.

**Reply by kvalev** [reply]
I had the same question - the immich geonames database is bundled on release - https://github.com/immich-app/immich/issues/11715#issuecomment-2324861497
